### ISSUE TITLE
How to create "pure" html pages (with no common.js, no common styles) from files ?

### ISSUE THREAD

**Issue by gael-boyenval** [gatsbyjs/gatsby#19188]
## Summary

I use gatsby to create a design system website.
In this website, I have a component that display elements inside an iframe. 

Why ?

- In iframe I can change the viewport, and see responsive changes
- I have a decoupled environment in which there is no collision between the "main" site CSS and the inside of the Iframe

<img width="796" alt="Capture d’écran 2019-10-31 à 16 46 03" src="https://user-images.githubusercontent.com/14542464/67964237-3885c080-fc00-11e9-8ebe-e4a4c17f4781.png">

Right now, I'm using the following method to render the iframe : 

- In specified directories, I have `.html` and `.scss` files
- Using a custom source plugin, I create a graphql node in which I store the html, scss, and css (compiled)
- using a static query, I feed a component and manually inject the dom and css inside the iframe.  

Not the cleanest thing, but It works. 

Right now, I would like, instead of manually injecting dom and CSS through a component, to create simple html pages and use the URL as the `src` attribute value of my iframe. 

The reason for that is I want to run automated visual testing on those pages. 
For that I need those pages to be serviced with their own dedicated URLS.

I managed to do it using the `createPages` function inside `gatsby-node.js`. 

But There is multiple problems with that : 

- Pages comes with code (js bundles, styles etc)
- As I call many of those iframes in each pages, it just break at some point.
- I can't manage the dom from top to bottom, I can just inject stuffs in the body

Is there a way, not too ugly preferably, to build those pages without injecting any react/gatsby/webpack related stuff inside ?

**Comment by LekoArts** [comment]
Thank you for opening this!

Gatsby currently has no options or abilities to create `pages without injecting any react/gatsby/webpack related stuff inside`. And we currently don't plan to change that or work on features that would enable that.

Maybe something like https://www.gatsbyjs.org/packages/gatsby-remark-embed-snippet/ is more helpful for you.

We're marking this issue as answered and closing it for now but please feel free to comment here if you would like to continue this discussion. We also recommend heading over to our [communities](https://www.gatsbyjs.org/contributing/community/#where-to-get-support) if you have questions that are not bug reports or feature requests. We hope we managed to help and thank you for using Gatsby!

**Comment by gusbemacbe** [comment]
@gael-boyenval and @LekoArts 

I migrated from Jekyll. Too bad you will not support it. :-( The directory gets junky, messful and uglier. 

Jekyll does a great job for generating pure HTML pages, but not totally complete as GatsbyJS. :-/

**Comment by gael-boyenval** [comment]
@gusbemacbe : 

The cleanest solution I found was the following : 

- I use NPM scripts to generate my html pages, that I send in public (under public/any-dir)
- then in Gatsby-node, I create a script that serve the compiled HTMLS as statics in dev mode : 

```js
const express = require('express');

exports.onCreateDevServer = ({ app }) => {
  app.use(express.static('my/path'));
};
```

then I use the filesystem source plugin to create the slugs (and required fields) within `onCreateNode` to access the data about the pure HTML files within my gatsby pages and component through graphql queries. 

hope it helps :)

**Comment by gusbemacbe** [comment]
@gael-boyenval 

>     * I use NPM scripts to generate my html pages, that I send in public (under public/any-dir)

Which NPM packages/scripts? Only `express`? (I have it installed before)

> then in Gatsby-node, I create a script that serve the compiled HTMLS as statics in dev mode :

I tested, using both `gatsby build` and `gatsby develop`, but your code did not give effect, I still saw many initial JS pages, many ugly `page-date` and folder `static`.

**Comment by gael-boyenval** [comment]
@gusbemacbe : 

I'm not sure we are talking about the same thing here. 

If your goal is not the same as mine, my solution will have no effect on gatsby generated files. (which works perfectly and do not require any changes in my opinion).

I used custom scripts, using the scripts key in the package.json... to compile files outside of Gatsby.

**Comment by gusbemacbe** [comment]
Hello @gael-boyenval 

My goal is that, I want to generate pure HTML pages with folders `assets/{css,imagens,javascript,fonts}`, like Jekyll does.

**Comment by gael-boyenval** [comment]
Use Jekill then ;)

**Comment by gael-boyenval** [comment]
But Gatsby is great, provide awesome optimizations and performance, with the power of react in the client.
Why would you care about what the output look like ? It's for the browser only.

**Comment by gusbemacbe** [comment]
Bceause I would give or sell my pojrects to common clients with basic and reasonable knoweldge of CSS, HTML and JavaScript. 

I migrated from Jekyll to Gatsby because Gatsby is based on React which is easy than pure JavaScript because of control arrays, flows and iterations in HTML inside a JSX file, so I did not have to write a lot and a lot of codes in pure HTML, taking a long time. I know HAML and YAML, but they are too limited in case of control arrays, flows and iterations.
